diff -up a/plugins/wacom/Makefile.am b/plugins/wacom/Makefile.am
--- a/plugins/wacom/Makefile.am
+++ b/plugins/wacom/Makefile.am
@@ -48,6 +48,7 @@ libgsdwacom_la_CPPFLAGS = \
 	$(AM_CPPFLAGS)
 
 libgsdwacom_la_CFLAGS = \
+	$(PLUGIN_CFLAGS)						\
 	$(SETTINGS_PLUGIN_CFLAGS)					\
 	$(LIBWACOM_CFLAGS)						\
 	$(AM_CFLAGS)
@@ -85,20 +86,23 @@ gsd_wacom_led_helper_LDFLAGS =				\
 
 gsd_wacom_led_helper_CFLAGS =				\
 	$(BACKLIGHT_HELPER_CFLAGS)
+else
+libexec_PROGRAMS =
 endif
 
 EXTRA_DIST += org.gnome.settings-daemon.plugins.wacom.policy.in.in
 
-noinst_PROGRAMS = test-wacom list-wacom test-devman test-multihead
+noinst_PROGRAMS = test-devman test-multihead
+libexec_PROGRAMS += gsd-test-wacom gsd-list-wacom
 
-test_wacom_SOURCES =							\
+gsd_test_wacom_SOURCES =						\
 	test-wacom.c							\
 	gsd-wacom-manager.c						\
 	gsd-wacom-manager.h						\
 	gsd-wacom-device.c						\
 	gsd-wacom-device.h
 
-test_wacom_CPPFLAGS = \
+gsd_test_wacom_CPPFLAGS = \
 	-I$(top_srcdir)/data/						\
 	-I$(top_srcdir)/gnome-settings-daemon				\
 	-I$(top_srcdir)/plugins/common					\
@@ -109,12 +113,13 @@ test_wacom_CPPFLAGS = \
 	-DLIBEXECDIR=\""$(libexecdir)"\"				\
 	$(AM_CPPFLAGS)
 
-test_wacom_CFLAGS =							\
+gsd_test_wacom_CFLAGS =							\
+	$(PLUGIN_CFLAGS)						\
 	$(SETTINGS_PLUGIN_CFLAGS)					\
 	$(LIBWACOM_CFLAGS)						\
 	$(AM_CFLAGS)
 
-test_wacom_LDADD = \
+gsd_test_wacom_LDADD = 							\
 	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la		\
 	$(top_builddir)/plugins/common/libcommon.la			\
 	$(builddir)/libgsdwacomhelpers.la				\
@@ -122,12 +127,12 @@ test_wacom_LDADD = \
 	$(LIBWACOM_LIBS)						\
 	$(NULL)
 
-list_wacom_SOURCES =							\
+gsd_list_wacom_SOURCES =						\
 	list-wacom.c							\
 	gsd-wacom-device.c						\
 	gsd-wacom-device.h
 
-list_wacom_CPPFLAGS = \
+gsd_list_wacom_CPPFLAGS = 						\
 	-I$(top_srcdir)/data/						\
 	-I$(top_srcdir)/gnome-settings-daemon				\
 	-I$(top_srcdir)/plugins/common					\
@@ -138,18 +143,19 @@ list_wacom_CPPFLAGS = \
 	-DGNOME_SETTINGS_LOCALEDIR=\""$(datadir)/locale"\"		\
 	$(AM_CPPFLAGS)
 
-list_wacom_CFLAGS =							\
+gsd_list_wacom_CFLAGS =							\
 	$(SETTINGS_PLUGIN_CFLAGS)					\
 	$(LIBWACOM_CFLAGS)						\
 	$(AM_CFLAGS)
 
-list_wacom_LDADD = \
+gsd_list_wacom_LDADD = \
 	$(top_builddir)/gnome-settings-daemon/libgsd-profile.la		\
 	$(top_builddir)/plugins/common/libcommon.la			\
 	$(builddir)/libgsdwacomhelpers.la				\
 	$(SETTINGS_DAEMON_LIBS)						\
 	$(SETTINGS_PLUGIN_LIBS)						\
-	$(LIBWACOM_LIBS)
+	$(LIBWACOM_LIBS)						\
+	-lm
 
 test_devman_SOURCES =							\
 	test-devman.c
