diff -up a/data/desktop_gnome_peripherals_wacom.schemas.in b/data/desktop_gnome_peripherals_wacom.schemas.in
--- a/data/desktop_gnome_peripherals_wacom.schemas.in
+++ b/data/desktop_gnome_peripherals_wacom.schemas.in
@@ -54,7 +54,7 @@
       <key>/schemas/desktop/gnome/peripherals/wacom/touch</key>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Wacom touch feature</short>
         <long>Enable this to only move the cursor when the user touches the tablet.</long>
diff -up a/plugins/wacom/gsd-wacom-manager.c b/plugins/wacom/gsd-wacom-manager.c
--- a/plugins/wacom/gsd-wacom-manager.c
+++ b/plugins/wacom/gsd-wacom-manager.c
@@ -618,11 +618,14 @@ set_wacom_settings (GsdWacomManager *man
 	set_absolute (device, gconf_value_get_bool (value));
 	gconf_value_free (value);
 
-	value = gsd_gconf_get (gsd_wacom_device_get_settings_path (device),
-			       gsd_wacom_device_get_schemas_path (device),
-			       KEY_AREA);
-	set_area (device, value);
-	gconf_value_free (value);
+	/* Ignore touch devices as they do not share the same range of values for area */
+	if (type != WACOM_TYPE_TOUCH) {
+		value = gsd_gconf_get (gsd_wacom_device_get_settings_path (device),
+				       gsd_wacom_device_get_schemas_path (device),
+				       KEY_AREA);
+		set_area (device, value);
+		gconf_value_free (value);
+	}
 
 	value = gsd_gconf_get (gsd_wacom_device_get_settings_path (device),
 			       gsd_wacom_device_get_schemas_path (device),
@@ -660,7 +663,8 @@ wacom_settings_changed (GConfValue     *
 			set_absolute (device, gconf_value_get_bool (value));
 	} else if (g_str_equal (key, KEY_AREA)) {
 		if (type != WACOM_TYPE_CURSOR &&
-		    type != WACOM_TYPE_PAD)
+		    type != WACOM_TYPE_PAD &&
+		    type != WACOM_TYPE_TOUCH)
 			set_area (device, value);
 	} else if (g_str_equal (key, KEY_DISPLAY)) {
 		if (type != WACOM_TYPE_CURSOR &&
