diff -up a/plugins/wacom/gsd-wacom-manager.c b/plugins/wacom/gsd-wacom-manager.c
--- a/plugins/wacom/gsd-wacom-manager.c
+++ b/plugins/wacom/gsd-wacom-manager.c
@@ -933,24 +933,35 @@ generate_key (GsdWacomTabletButton *wbut
 	/* Couldn't find it in the current group? Look in group 0 */
 	if (keycode == 0) {
 		for (i = 0; i < n_keys; i++) {
-			if (keys[i].level > 0)
+			if (keys[i].group > 0)
 				continue;
 			keycode = keys[i].keycode;
 		}
 	}
 	g_free (keys);
 
-	g_debug ("Emitting '%s' (keyval: %d, keycode: %d mods: 0x%x)",
-		 str, keyval, keycode, mods);
-	g_free (str);
+	if (keycode == 0) {
+		g_warning ("Not emitting '%s' (keyval: %d, keycode: %d mods: 0x%x), invalid keycode",
+			   str, keyval, keycode, mods);
+		g_free (str);
+		return;
+	} else {
+		g_debug ("Emitting '%s' (keyval: %d, keycode: %d mods: 0x%x)",
+			 str, keyval, keycode, mods);
+	}
 
 	/* And send out the keys! */
+	gdk_error_trap_push ();
 	if (is_press)
 		send_modifiers (display, mods, TRUE);
 	XTestFakeKeyEvent (display, keycode,
 			   is_press ? True : False, 0);
 	if (is_press == FALSE)
 		send_modifiers (display, mods, FALSE);
+	if (gdk_error_trap_pop ())
+		g_warning ("Failed to generate fake key event '%s'", str);
+
+	g_free (str);
 }
 
 static GdkFilterReturn
